### PR TITLE
ORC-839: Fix head command for batch reader

### PR DESCRIPTION
**Pull request by yuokada**
### What changes were proposed in this pull request?

- Support a feature like `head` command in `Java Tools`

### Why are the changes needed?

- Bugfix for ORC-728

### How was this patch tested?

```
~/w/I/o/java ❯❯❯ java -jar tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar  data -n 1 ../examples/orc-file-11-format.orc
log4j:WARN No appenders could be found for logger (org.apache.hadoop.util.Shell).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.hadoop.security.authentication.util.KerberosUtil (file:/Users/yuokada/works/IdeaProjects/orc/java/tools/target/orc-tools-1.7.0-SNAPSHOT-uber.jar) to method sun.security.krb5.Config.getInstance()
WARNING: Please consider reporting this to the maintainers of org.apache.hadoop.security.authentication.util.KerberosUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Processing data file ../examples/orc-file-11-format.orc [length: 373336]
{"boolean1":false,"byte1":1,"short1":1024,"int1":65536,"long1":9223372036854775807,"float1":1,"double1":-15,"bytes1":[0,1,2,3,4],"string1":"hi","middle":{"list":[{"int1":1,"string1":"bye"},{"int1":2,"string1":"sigh"}]},"list":[{"int1":3,"string1":"good"},{"int1":4,"string1":"bad"}],"map":[],"ts":"2000-03-12 15:00:00.0","decimal1":"12345678.6547456"}
________________________________________________________________________________________________________________________

```

### CODE REVIEW

File: java/tools/src/java/org/apache/orc/tools/PrintData.java [hunk 1]
```diff
@@ -207,6 +207,7 @@ static void printJsonData(PrintStream printStream,
       TypeDescription schema = reader.getSchema();
       VectorizedRowBatch batch = schema.createRowBatch();
       Integer counter = 0;
+      label:
```
Comment (pgaref) [inline java/tools/src/java/org/apache/orc/tools/PrintData.java:210]: Can we change the logic here avoiding labels?
Want to make sure readability here

**Comment by pgaref** [comment]
@yuokada  thanks for the FIX!
ORC-728 is closed so we will need a new one -- can you please create one and update PR?

**Comment by yuokada** [comment]
@pgaref Thanks for your comment. I filed ORC-839.

File: java/tools/src/java/org/apache/orc/tools/PrintData.java (hunk 1 above)
Comment (yuokada) [reply java/tools/src/java/org/apache/orc/tools/PrintData.java]: @pgaref I will try to change (I changed the status of this to DRAFT). But, I think that the 1st code is the simplest way to fix this issue.

**Comment by dongjoon-hyun** [comment]
Thank you for updating, @yuokada . If this is ready for review, could you convert your PR status?

**Comment by dongjoon-hyun** [comment]
cc @williamhyun

**Comment by williamhyun** [comment]
+1 LGTM
Merged to main.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @yuokada and @williamhyun .

**Pull request closed without merging**